fs-cache/src/cache.rs

```diff
@@ -0,0 +1,668 @@
+    /// * `max_memory_bytes` - Maximum bytes to keep in memory
+    /// * `log_values` - Whether to include values in debug logs
+    pub fn new(
+        label: String,
+        path: &Path,
+        max_memory_bytes: usize,
+        log_values: bool,
+    ) -> Result<Self> {
+        let mut cache = Self {
+            label: label.clone(),
+            path: PathBuf::from(path),
+            // TODO: NEED FIX
+            memory_cache: LruCache::new(
+                NonZeroUsize::new(max_memory_bytes)
+                    .expect("Capacity can't be zero"),
+            ),
+            current_memory_bytes: 0,
+            max_memory_bytes,
+            log_values,
+        };
+
+        log::debug!(
+            "cache/{}: initialized with {} bytes limit",
+            label,
+            max_memory_bytes
+        );
+
+        cache.load_fs()?;
+        Ok(cache)
+    }
+
+    /// Retrieves a value by its key, checking memory first then disk.
+    /// Returns None if the key doesn't exist.
+    pub fn get(&mut self, key: &K) -> Option<V> {
+        log::debug!("cache/{}: retrieving value for key {}", self.label, key);
+
+        let value = self
+            .fetch_from_memory(key)
+            .or_else(|| self.fetch_from_disk(key));
+
+        match &value {
+            Some(v) => {
+                if self.log_values {
+                    log::debug!(
+                        "cache/{}: found value for key {}: {:?}",
+                        self.label,
+                        key,
+                        v
+                    );
+                }
+            }
+            None => {
+                log::warn!(
+                    "cache/{}: no value found for key {}",
+                    self.label,
+                    key
+                );
+            }
+        }
+
+        value
+    }
+
+    /// Stores a new value with the given key.
+    /// Returns error if the key already exists or if writing fails.
+    pub fn set(&mut self, key: K, value: V) -> Result<()> {
+        log::debug!("cache/{}: setting value for key {}", self.label, key);
+
+        // Check if value already exists
+        if self.exists(&key) {
+            return Err(ArklibError::Storage(
+                self.label.clone(),
+                format!("Key {} already exists in cache", key),
+            ));
+        }
+
+        // Always write to disk first
+        self.persist_to_disk(&key, &value)?;
+
+        // Then update memory cache
+        self.update_memory_cache(&key, &value)?;
+
+        log::debug!("cache/{}: set key={}", self.label, key);
+        Ok(())
+    }
+
+    /// Checks if a value exists either in memory or on disk.
+    pub fn exists(&self, key: &K) -> bool {
+        self.memory_cache.contains(key)
+            || self.path.join(key.to_string()).exists()
+    }
+
+    /// Returns an ordered iterator over all cached keys.
+    pub fn keys(&self) -> Result<impl Iterator<Item = K>> {
```

Comment:
How is the iterator ordered? Is it by key, by insertion order, or something else? I don’t see any specific ordering in the code.